libs/dal/src/repositories/message/message.schema.ts

```diff
@@ -132,6 +127,17 @@ messageSchema.virtual('template', {
 
 messageSchema.plugin(mongooseDelete, { deletedAt: true, deletedBy: true, overrideMethods: 'all' });
 
+messageSchema.index({
```

Comment:
Regarding the order of the index. As mentioned [here](https://books.google.co.il/books?id=uGUKiNkKRJ0C&pg=PA98&lpg=PA98&dq=mongodb+the+definitive+guide+index+cardinality&source=bl&ots=h8mzPdcTrl&sig=xJ40rE2qjoSy9RQrvxLnNDx6KS8&hl=en&sa=X&redir_esc=y#v=onepage&q=mongodb%20the%20definitive%20guide%20index%20cardinality&f=false).
the rule of thumb is to put the high-cardinality keys first in compound indeces before the low-cardinality ones.